rtrlib/pfx/trie/trie-pfx.c

```diff
@@ -133,18 +153,31 @@ inline struct trie_node *pfx_table_get_root(const struct pfx_table *pfx_table, c
 
 int pfx_table_del_elem(struct node_data *data, const unsigned int index)
 {
+    struct data_elem *tmp;
+    struct data_elem deleted_elem = data->ary[index];
```

Comment:
where is this ensured? And if thats the case, we should have an `assert((data->len == 0) || (index < date->len))` here at least, or not?